components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/model/AuthenticatedUser.java

```diff
@@ -411,7 +411,8 @@ public void setAuthenticatedSubjectIdentifier(String authenticatedSubjectIdentif
                 authenticatedSubjectIdentifier = IdentityUtil.addDomainToName(userName, userStoreDomain);
             }
             if (useTenantDomainInLocalSubjectIdentifier && StringUtils.isNotEmpty(tenantDomain) &&
-                    StringUtils.isNotEmpty(authenticatedSubjectIdentifier)) {
+                    StringUtils.isNotEmpty(authenticatedSubjectIdentifier) &&
+                    StringUtils.countMatches(authenticatedSubjectIdentifier, "@") < 2) {
```

Comment:
use tenant separator constant instead of '@'

Reply:
Addressed in https://github.com/wso2/carbon-identity-framework/pull/4851/commits/a4ba7011ef9323d90540fcba1746bf86d8f1f0ac